### PR TITLE
Number of Items Issue Solved

### PR DESCRIPTION
**Pull request by ranjankuldeep**
#92 

# The Cart is working as expected now.
## Here is the ScreenShot showing correct number of items.
I have tested many possibilities of adding to the cart and editing in the cart, Everythings work fine.

 
Screenshot 2022-10-12 at 4 10 54 PM

@roopeshsn

### CODE REVIEW

File: frontend/src/screens/CartScreen.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const CartScreen = ({ match, location, history }) => {
   const { cartItems } = cart
 
   const cartItemsPrice = cartItems.reduce(
-    (acc, item) => acc + item.qty * item.price,
+    (acc, item) => acc + +item.qty * +item.price,
```
Comment (roopeshsn) [inline frontend/src/screens/CartScreen.js:26]: What did this plus sign (+) do?

**Review by roopeshsn — CHANGES_REQUESTED** [review]

File: frontend/src/screens/CartScreen.js (hunk 1 above)
Comment (ranjankuldeep) [reply frontend/src/screens/CartScreen.js:26]: The unary plus operator (+) precedes its operand and evaluates to its operand but attempts to convert it into a number, if it isn't already.

It basically makes string to number.
So these are now no longer added as string but as a number.
@roopeshsn

Comment (roopeshsn) [reply frontend/src/screens/CartScreen.js:26]: Why have you not used a function? It will improve readability instead of using the unary plus operator.

Comment (ranjankuldeep) [reply frontend/src/screens/CartScreen.js:26]: We can also go with the parseInt(),
But i choosen '+' operator here.

Comment (ranjankuldeep) [reply frontend/src/screens/CartScreen.js:26]: Ohhk so I will make the changes.
Will be using function now .

**Comment by roopeshsn** [comment]
Also, format the code using the prettier command. Check the package.json file.

**Comment by ranjankuldeep** [comment]
Please Review,
I have made the necessary changes. @roopeshsn

File: frontend/src/screens/CartScreen.js [hunk 2]
```diff
@@ -79,7 +79,7 @@ const CartScreen = ({ match, location, history }) => {
                       <Form.Control
                         as="select"
                         size="sm"
-                        value={item.qty}
+                        value={+item.qty}
```
Comment (roopeshsn) [inline frontend/src/screens/CartScreen.js:82]: Why have you not changed here?

File: frontend/src/screens/CartScreen.js [hunk 3]
```diff
@@ -116,7 +116,7 @@ const CartScreen = ({ match, location, history }) => {
           <ListGroup variant="flush">
             <ListGroup.Item>
               <h5>
-                Subtotal ({cartItems.reduce((acc, item) => acc + item.qty, 0)})
+                Subtotal ({cartItems.reduce((acc, item) => acc + +item.qty, 0)})
```
Comment (roopeshsn) [inline frontend/src/screens/CartScreen.js:119]: Here too

**Review by roopeshsn — CHANGES_REQUESTED** [review]

File: frontend/src/screens/CartScreen.js (hunk 3 above)
Comment (ranjankuldeep) [reply frontend/src/screens/CartScreen.js]: Ahh,  I missed that.

**Comment by roopeshsn** [comment]
Whether there the changes didn't affect the working of the application right? Have you formatted the code using prettier?

**Comment by ranjankuldeep** [comment]
I made the changes in the cart screen. 
@roopeshsn

**Comment by roopeshsn** [comment]
> Whether there the changes didn't affect the working of the application right? Have you formatted the code using prettier?

Reply appropriately!

**Comment by ranjankuldeep** [comment]
yes i formatted ,
i run the command 
```
npm run prettier 
```
I didn't get the first question.

**Comment by ranjankuldeep** [comment]
The application is working fine now.
I added changes only where quantity was involved so that it won't break at any point

**Comment by roopeshsn** [comment]
Man answer in a single shot. What about prettier?

**Pull request merged**